phylofisher/matrix_constructor.smk

```diff
@@ -20,88 +20,89 @@ out_dict = {'fasta':          'fas',
-    log:
-        f'{out_dir}/logs/divvier/{{gene}}.log'
-    conda:
-        'divvier.yaml'
-    shell:
-        f'''
-        divvier -mincol 4 -partial -divvygap {{input}} >{{log}} 2>{{log}}
-
-        mv {out_dir}/mafft/{{wildcards.gene}}.aln.partial.fas {out_dir}/divvier >{{log}} 2>{{log}}
-        mv {out_dir}/mafft/{{wildcards.gene}}.aln.PP {out_dir}/divvier >{{log}} 2>{{log}}
-        '''
-
-rule trimal:
+    rule prequal:
         input:
-            f'{out_dir}/divvier/{{gene}}.aln.partial.fas'
+            f'{out_dir}/prequal/{{gene}}.aa'
         output:
-            f'{out_dir}/trimal/{{gene}}.trimal'
+            f'{out_dir}/prequal/{{gene}}.aa.filtered'
         log:
-            f'{out_dir}/logs/trimal/{{gene}}.log'
+            f'{out_dir}/logs/prequal/{{gene}}.log'
         conda:
-            'trimal.yaml'
-        params:
-            gt=trimal_gt
+            'prequal.yaml'
         shell:
-            'trimal -in {input} -gt {params.gt} -out {output} >{log} 2>{log}'
+            'prequal {input} >{log} 2>{log}'
 
 
-rule remove_gaps:
+    rule mafft:
         input:
-            f'{out_dir}/trimal/{{gene}}.trimal'
+            f'{out_dir}/prequal/{{gene}}.aa.filtered'
         output:
-            f'{out_dir}/trimal/{{gene}}.final'
+            f'{out_dir}/mafft/{{gene}}.aln'
         log:
-            f'{out_dir}/logs/trimal/{{gene}}.log'
-        run:
-            records = []
-            for record in SeqIO.parse(input[0], 'fasta'):
-                if len(str(record.seq).replace('-', '').replace('X','')) > 0:
-                    records.append(record)
+            f'{out_dir}/logs/mafft/{{gene}}.log'
+        conda:
+            'mafft.yaml'
+        shell:
+            'mafft --thread 1 --globalpair --maxiterate 1000 --unalignlevel 0.6 {input} >{output} 2>{log}'
 
-            SeqIO.write(records, output[0], "fasta")
+    rule divvier:
+        input:
+            f'{out_dir}/mafft/{{gene}}.aln'
+        output:
+            f'{out_dir}/divvier/{{gene}}.aln.partial.fas',
+            f'{out_dir}/divvier/{{gene}}.aln.PP'
+        log:
+            f'{out_dir}/logs/divvier/{{gene}}.log'
+        conda:
+            'divvier.yaml'
+        shell:
+            f'''
+            divvier -mincol 4 -partial -divvygap {{input}} >{{log}} 2>{{log}}
+
+            mv {out_dir}/mafft/{{wildcards.gene}}.aln.partial.fas {out_dir}/divvier >{{log}} 2>{{log}}
+            mv {out_dir}/mafft/{{wildcards.gene}}.aln.PP {out_dir}/divvier >{{log}} 2>{{log}}
+            '''
+
+    rule trimal:
+            input:
+                f'{out_dir}/divvier/{{gene}}.aln.partial.fas'
+            output:
+                f'{out_dir}/trimal/{{gene}}.trimal'
+            log:
+                f'{out_dir}/logs/trimal/{{gene}}.log'
+            conda:
+                'trimal.yaml'
+            params:
+                gt=trimal_gt
+            shell:
+                'trimal -in {input} -gt {params.gt} -out {output} >{log} 2>{log}'
```

Comment:
The indent should be fine because it's still underneath `if not concatenation_only:`. I think it just looks weird because it is so far away from the if statement.